custom_components/sagemcom_fast/__init__.py

```diff
@@ -101,6 +101,11 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
         _LOGGER.exception(exception)
         return False
 
+    try:
```

Comment:
I don't think that a try catch here is required. This will cause it to fail later, when gateway cannot be retrieved.

Reply:
That would mean it stays logged in, which causes problems when the next login is tried.
So, I think it is.

Reply:
You are right, I missed that you did a try / finally and not a try / catch.